File: frontend/src/shared/tool-registry/registry.ts
export interface ToolMeta {
  id: string
  name: string
  description: string
  path: string
}

export const TOOL_REGISTRY: ToolMeta[] = [
  {
    id: 'srt-split',
    name: 'Tách SRT',
    description: 'Chia file SRT gốc theo thời lượng từng episode video',
    path: '/tools/srt-split',
  },
]

File: backend/app/tools/registry.py
from dataclasses import dataclass
from typing import Type

import customtkinter as ctk

from app.tools.srt_split.gui import SrtSplitFrame


@dataclass
class ToolMeta:
    id: str
    name: str
    description: str
    frame_class: Type[ctk.CTkFrame]


TOOL_REGISTRY: list[ToolMeta] = [
    ToolMeta(
        id="srt-split",
        name="Tách SRT",
        description="Chia file SRT gốc theo thời lượng từng episode video",
        frame_class=SrtSplitFrame,
    ),
]
